spec/controllers/admin/user_creations_controller_spec.rb

```diff
@@ -0,0 +1,106 @@
+  include LoginMacros
+  include RedirectExpectationHelper
+
+  describe "GET #hide" do    
+    let(:admin) { create(:admin) }
+    let(:work) { create(:work) }
+
+    context 'when admin does not have correct authorization' do
+      it "denies random admin access" do
+        admin.update(roles: [])
+        fake_login_admin(admin)
+        get :hide, params: { id: work.id, creation_type: 'Work' }
+        it_redirects_to_with_error(root_url, "Sorry, only an authorized admin can access the page you were trying to reach.")
+      end
+    end
+
+    context 'when admin does have correct authorization' do
+      it "allows admin with authorization to hide user_creation" do
+        admin.update(roles: ['policy_and_abuse'])
+        fake_login_admin(admin)
+        get :hide, params: { id: work.id, creation_type: 'Work', hidden: true }
+
+        it_redirects_to_with_notice(work_path(work), "Item has been hidden.")
+        work.reload
+        expect(work.hidden_by_admin).to eq(true)
+      end
+
+      it "allows admin with authorization to make user_creation visible" do
+        work.update(hidden_by_admin: true)
+        admin.update(roles: ['policy_and_abuse'])
+        fake_login_admin(admin)
+        get :hide, params: { id: work.id, creation_type: 'Work', hidden: false }
+
+        it_redirects_to_with_notice(work_path(work), "Item is no longer hidden.")
+        work.reload
+        expect(work.hidden_by_admin).to eq(false)
+      end
+    end
+  end
+
+  describe "GET #set_spam" do
+    let(:admin) { create(:admin) }
+    let(:work) { create(:work) }
+
+    context 'when admin does not have correct authorization' do
+      it "denies random admin access" do
+        admin.update(roles: [])
+        fake_login_admin(admin)
+        get :set_spam, params: { id: work.id, creation_type: 'Work', spam: true }
+        it_redirects_to_with_error(root_url, "Sorry, only an authorized admin can access the page you were trying to reach.")
+      end
+    end
+
+    context 'when admin does have correct authorization' do
+      it "allows admin with authorization to mark user_creation as spam" do
+        admin.update(roles: ['policy_and_abuse'])
+        fake_login_admin(admin)
+        get :set_spam, params: { id: work.id, creation_type: 'Work', spam: true }
+
+        it_redirects_to_with_notice(work_path(work), "Work was marked as spam and hidden.")
+        work.reload
+        expect(work.spam).to eq(true)
+      end
+
+      it "allows admin with authorization to mark user_creation as NOT spam" do
+        admin.update(roles: ['policy_and_abuse'])
+        work.update(spam: true)
+        fake_login_admin(admin)
+        get :set_spam, params: { id: work.id, creation_type: 'Work', spam: false }
+
+        it_redirects_to_with_notice(work_path(work), "Work was marked not spam and unhidden.")
+        work.reload
+        expect(work.spam).to eq(false)
+      end
+    end
+  end
+
+  describe "GET #destroy" do
```

Comment:
I think this is a `DELETE` rather than a `GET`, but I'm actually not 100% sure.

Reply:
That's how `rake routes` expands it, yeah.
`
admin_user_creation DELETE /admin/user_creations/:id(.:format)  admin/user_creations#destroy`